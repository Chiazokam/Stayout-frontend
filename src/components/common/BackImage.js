import React from 'react';

const BackImage = (props) => {
  const { className } = props;

  return (
    <section className={className} />
  );
};

export default BackImage;
