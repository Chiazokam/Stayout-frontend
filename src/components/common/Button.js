import React from 'react';
import { Row, Col } from 'reactstrap';


const Button = (props) => {
  const { className, children } = props;

  return (
    <button
      className={className}
      type='submit'
    >
      {children}
    </button>
  );
};

export default Button;
