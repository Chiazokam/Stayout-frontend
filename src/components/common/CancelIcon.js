import React from 'react';
import cancel from '../../assets/images/cancel.svg';

const CancelIcon = (props) => {
  const { clicked } = props;

  return (
    <div>
      <img
        className='cancel-icon'
        onClick={clicked}
        src={cancel}
        alt='cancel'
        role='presentation'
      />
    </div>
  );
};

export default CancelIcon;
