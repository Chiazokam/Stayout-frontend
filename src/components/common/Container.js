import React from 'react';

const Container = (props) => {
  const { children } = props;
  return (
    <div className='container'>
      {children}
    </div>
  );
};

export default Container;
