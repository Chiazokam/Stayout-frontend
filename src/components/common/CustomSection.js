import React from 'react';


const CustomSection = (props) => {
  const { className, children } = props;

  return (
    <div className={className}>
      {children}
    </div>
  );
};

export default CustomSection;
