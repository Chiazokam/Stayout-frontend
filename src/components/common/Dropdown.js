import React, { Component } from 'react';

const Dropdown = (props) => {
  const { className, menu } = props;
  return (
    <nav className={`dropdown ${className}`}>
      <ul>
        {menu.map(menuItem => (
          <li key={menuItem}>
            {menuItem}
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default Dropdown;
