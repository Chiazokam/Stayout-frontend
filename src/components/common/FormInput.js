import React from 'react';
import {
  FormGroup,
  Label,
  Input,
  InputGroup,
  InputGroupAddon,
  InputGroupText
} from 'reactstrap';

const FormInput = (props) => {
  const {
    label, type, name, placeholder, labelFor, className, addonText, value, onChange, disabled
  } = props;

  return (
    <FormGroup>
      <InputGroup>
        <Label for={labelFor}>{label}</Label>
        <Input
          className={className}
          type={type}
          name={name}
          placeholder={placeholder}
          value={value}
          onChange={onChange}
          disabled={disabled}
        />
        {addonText && (
        <InputGroupAddon addonType='append'>
          <InputGroupText>{addonText}</InputGroupText>
        </InputGroupAddon>
        )}
      </InputGroup>
    </FormGroup>
  );
};

export default FormInput;
