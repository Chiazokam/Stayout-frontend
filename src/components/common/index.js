import BackImage from './BackImage';
import Logo from './Logo';
import Container from './Container';
import MobileHeader from './MobileHeader';
import Hamburger from './Hamburger';
import Button from './Button';
import CustomSection from './CustomSection';
import ThinLine from './ThinLine';
import Dropdown from './Dropdown';

export {
  BackImage,
  Logo,
  Container,
  MobileHeader,
  Hamburger,
  Button,
  CustomSection,
  ThinLine,
  Dropdown
};
