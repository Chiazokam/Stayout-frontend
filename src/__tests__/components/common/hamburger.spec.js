import React from 'react';
import { shallow } from 'enzyme';
import { Hamburger } from '../../../components/common';

describe('Hamburger Tests', () => {
  it('should render correctly', () => {
    const tree = shallow(<Hamburger />);
    expect(tree).toMatchSnapshot();
  });
});
