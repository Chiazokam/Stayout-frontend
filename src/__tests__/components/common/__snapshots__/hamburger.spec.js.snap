// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hamburger Tests should render correctly 1`] = `ShallowWrapper {}`;
